workflow-bot-app/src/test/java/com/symphony/bdk/workflow/FormReplyIntegrationTest.java

```diff
@@ -323,4 +322,53 @@ void sendMessageUpdateMessage() throws Exception {
       assertThat(workflow).executed("init", "message-received_/hey", "update");
     });
   }
+
+  @Test
+  void formRepliedSendMessageOnConditionIf() throws Exception {
+    Workflow workflow =
+        SwadlParser.fromYaml(getClass().getResourceAsStream("/form/send-form-reply-conditional.swadl.yaml"));
+
+    when(messageService.send(anyString(), any(Message.class))).thenReturn(message("msgId"));
+
+    engine.deploy(workflow);
+
+    // trigger workflow execution
+    engine.onEvent(messageReceived("/test"));
+    verify(messageService, timeout(5000)).send(anyString(), contains("form"));
+
+    await().atMost(5, TimeUnit.SECONDS).ignoreExceptions().until(() -> {
+      engine.onEvent(form("msgId", "testForm", Collections.singletonMap("action", "create")));
+      return true;
+    });
+    verify(messageService, timeout(5000)).send(anyString(), contains("Create"));
```

Comment:
I'm not sure this verifies what you think. 
In line 337, we verify that we sent the form (content contains the word form). 
In line 343, we want to verify that we sent a the message "Create". I don't know why Mockito.verify does not get updated, and it gets applied on the first one. So both Mockito.verify applies on the first messageService.send(..), which contains both "form" and "Create" words. 

If you change the activity "resCreate" in swadl to send something else than "Create", let's say "CreateABC", and you change the test with (contains("CreateABC")), the test will fail.. One way to force Mockito.verify to apply to the new sendMessage call, is to change the to.streamId="" to a different one and replace anyString() in verify(..). 


```verify(messageService, timeout(5000)).send("ANOTHER_STREAM_ID", contains("SOMETHING_DIFF_FROM_CREATE"));```

Reply:
ok, i see your point, i add then the Mockito.clearInvocations(T mock) to clear the invocations on messageService mock after each verify, so that we are sure we have the right verification.